signer/jws_test.go

```diff
@@ -0,0 +1,148 @@
+package signer
+
+import (
+	"crypto/x509"
+	"errors"
+	"github.com/notaryproject/notation-core-go/internal/testhelper"
+	"testing"
+)
+
+func TestNewJWSEnvelopeFromBytes(t *testing.T) {
+	t.Run("newJWSEnvelopeFromBytes", func(t *testing.T) {
+		if _, err := newJWSEnvelopeFromBytes([]byte(TestValidSig)); err != nil {
+			t.Errorf("Error found")
+		}
+	})
+
+	t.Run("newJWSEnvelopeFromBytes Error", func(t *testing.T) {
+		if _, err := newJWSEnvelopeFromBytes([]byte("Malformed")); err == nil {
+			t.Errorf("Expected error but not found")
+		}
+	})
+}
+
+func TestValidateIntegrity(t *testing.T) {
+	t.Run("with newJWSEnvelope() returns error", func(t *testing.T) {
+		env := jwsEnvelope{}
+		err := env.validateIntegrity()
+		if !(err != nil && errors.As(err, new(SignatureNotFoundError))) {
+			t.Errorf("Expected SignatureNotFoundError but found %q", err)
+		}
+	})
+
+	t.Run("with NewJWSEnvelopeFromBytes works", func(t *testing.T) {
+		env, _ := newJWSEnvelopeFromBytes([]byte(TestValidSig))
+		err := env.validateIntegrity()
+		if err != nil {
+			t.Errorf("validateIntegrity(). Error = %s", err)
+		}
+	})
+
+	t.Run("with invalid base64 bytes sig envelope returns error", func(t *testing.T) {
+		env, _ := newJWSEnvelopeFromBytes([]byte("{\"Payload\":\"Hi!\",\"Protected\":\"Hi\",\"Header\":{},\"Signature\":\"Hi!\"}"))
+		if err := env.validateIntegrity(); err == nil {
+			t.Errorf("Expected error but not found")
+		}
+	})
+
+	t.Run("with incomplete sig envelope returns error", func(t *testing.T) {
+		env, _ := newJWSEnvelopeFromBytes([]byte("{\"Payload\":\"eyJhbGciOiJIUzI1NiJ9\",\"Protected\":\"eyJhbGciOiJQUzI1NiIsImNyaXQiOlsiaW8uY25jZi5ub3Rhcnkuc2lnbmluZ1RpbWUiXSwiaW8uY25jZi5ub3Rhcnkuc2luaW5nVGltZSI6IjIwMDYtMDEtMDJUMTU6MDQ6MDVaIn0\",\"Header\":{},\"Signature\":\"YjGj\"}"))
+		if err := env.validateIntegrity(); !(err != nil && errors.As(err, new(MalformedSignatureError))) {
+			t.Errorf("Expected MalformedSignatureError but but found %q", err)
+		}
+	})
+}
+
+func TestGetSignerInfo(t *testing.T) {
+	t.Run("with newJWSEnvelope before sign returns error", func(t *testing.T) {
+		env := jwsEnvelope{}
+		_, err := env.getSignerInfo()
+		if !(err != nil && errors.As(err, new(SignatureNotFoundError))) {
+			t.Errorf("Expected SignatureNotFoundError but found %q", err)
+		}
+	})
+
+	t.Run("with newJWSEnvelope after sign sign works", func(t *testing.T) {
+		env := jwsEnvelope{}
+		_, err := env.getSignerInfo()
+		env.signPayload(getSignRequest())
+		if !(err != nil && errors.As(err, new(SignatureNotFoundError))) {
+			t.Errorf("getSignerInfo(). Error = %s", err)
```

Comment:
You can use `t.Name()` in the error messages instead of hard coding the method names you are testing. That way, it is easier for devs to find a failing unit test.